lib/evaluation/predicateRelatedFieldEvaluationHandler.ts

```diff
@@ -0,0 +1,66 @@
+import { Constraints, Optionality, PresentationDefinition } from '@sphereon/pe-models';
+
+import { Status } from '../ConstraintUtils';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class PredicateRelatedFieldEvaluationHandler extends AbstractEvaluationHandler {
+  public getName(): string {
+    return 'PredicateRelatedField';
+  }
+
+  public handle(pd: PresentationDefinition, _p: unknown, results: HandlerCheckResult[]): void {
+    for (let i = 0; i < pd.input_descriptors.length; i++) {
+      const constraints: Constraints = pd.input_descriptors[i].constraints;
+      if (constraints) {
+        this.examinePredicateRelatedField(i, constraints, results);
+      }
+    }
+  }
+
+  private examinePredicateRelatedField(
+    input_descriptor_idx: number,
+    constraints: Constraints,
+    results: HandlerCheckResult[]
+  ): void {
+    for (let i = 0; i < constraints.fields.length; i++) {
+      for (let j = 0; j < results.length; j++) {
+        const resultInputDescriptorIdx = this.retrieveResultInputDescriptorIdx(results[j].input_descriptor_path);
+        if (results[j].evaluator !== 'FilterEvaluation' || input_descriptor_idx !== resultInputDescriptorIdx) {
+          continue;
+        }
+        if (constraints.fields[i].predicate) {
+          if (constraints.fields[i].predicate === Optionality.Required) {
+            results.push({
+              input_descriptor_path: `$.input_descriptors[${input_descriptor_idx}]`,
+              verifiable_credential_path: results[j].verifiable_credential_path,
+              evaluator: this.getName(),
+              status: Status.INFO,
+              message: 'Input candidate valid for presentation submission',
+              payload: results[j].payload.result.value,
+            });
+          } else {
+            results.push({
+              input_descriptor_path: `$.input_descriptors[${input_descriptor_idx}]`,
+              verifiable_credential_path: results[j].verifiable_credential_path,
+              evaluator: this.getName(),
+              status: Status.INFO,
+              message: 'Input candidate valid for presentation submission',
+              payload: true,
+            });
+          }
+        }
+      }
+    }
+  }
```

Comment:
Can be subdevided using extract-method-refactoring available in the IntelliJ
It is prefferred to not use continue.
It is prefferred to have functions with less than 3 Cyclomatic Complexity

Reply:
done